be/src/exprs/time_functions.cpp

```diff
@@ -2361,4 +2361,125 @@ StatusOr<ColumnPtr> TimeFunctions::make_date(FunctionContext* context, const Col
 
     return result.build(ColumnHelper::is_all_const(columns));
 }
+
+// last_day
+StatusOr<ColumnPtr> TimeFunctions::last_day(FunctionContext* context, const Columns& columns) {
+    ColumnViewer<TYPE_DATETIME> data_column(columns[0]);
+    auto size = columns[0]->size();
+
+    ColumnBuilder<TYPE_DATE> result(size);
+    for (int row = 0; row < size; ++row) {
+        if (data_column.is_null(row)) {
+            result.append_null();
+            continue;
+        }
+
+        DateValue date = (DateValue)data_column.value(row);
+        date.set_end_of_month(); // default month
+        result.append(date);
+    }
+    return result.build(ColumnHelper::is_all_const(columns));
+}
+
+StatusOr<ColumnPtr> TimeFunctions::_last_day_with_format_const(std::string& format_content, FunctionContext* context,
+                                                               const Columns& columns) {
+    ColumnViewer<TYPE_DATETIME> data_column(columns[0]);
+    auto size = columns[0]->size();
+
+    ColumnBuilder<TYPE_DATE> result(size);
+    for (int row = 0; row < size; ++row) {
+        if (data_column.is_null(row)) {
+            result.append_null();
+            continue;
+        }
+
+        DateValue date = (DateValue)data_column.value(row);
+        if (format_content == "year") {
+            date.set_end_of_year();
+        } else if (format_content == "quarter") {
+            date.set_end_of_quarter();
+        } else {
+            date.set_end_of_month();
+        }
+        result.append(date);
+    }
+    return result.build(ColumnHelper::is_all_const(columns));
+}
+
+StatusOr<ColumnPtr> TimeFunctions::_last_day_with_format_optional(FunctionContext* context, const Columns& columns) {
```

Comment:
optional means something is present or absent,  here the format argument is absolutely present , so can we just use   name  "_last_day_with_format" to indicates it is a default and non-optimized branch of last_day?

Reply:
ok, I will last_day_with_format_optional modify _last_day_with_format